src/main/kotlin/uk/gov/justice/hmpps/offendersearch/config/ElasticSearchConfiguration.kt

```diff
@@ -22,16 +30,37 @@ class ElasticSearchConfiguration {
   private val shouldSignRequests = false
   @Value("\${aws.region:eu-west-2}")
   private val awsRegion: String? = null
+  @Value("\${aws.roleArn:test-role-arn}")
+  private val roleARN: String? = null
+  @Value("\${aws.roleSessionName:test-role-session}")
+  private val roleSessionName: String? = null
 
   @Bean
   fun elasticSearchClient(): RestHighLevelClient {
     if (shouldSignRequests) {
+
+      val stsClient: AWSSecurityTokenService = AWSSecurityTokenServiceClientBuilder.standard()
```

Comment:
This role assumption code may or may not want to be inside the if (shouldSignRequests). Or the code to create the credentials may want pulling out into its own method so it's easier to change the point at which that bit is done